src/decoder/mod.rs

```diff
@@ -635,6 +641,21 @@ impl<R: BufRead + Seek> Reader<R> {
         size * height as usize
     }
 
+    /// Return the number of bytes required to hold a deinterlaced image frame, if it fits into the
+    /// memory space of the machine.
+    pub fn checked_output_buffer_size(&self) -> Option<usize> {
```

Comment:
The function could also return an error, LimitExceeded is arguably correct. @Shnatsel any preference on this? (Aside: I'll do the same transform to output_line_size afterall, might as well strive for consistency in the API break).